### PR TITLE
[ML-DataFrame] migrate to PageParams for get and stats, move PageParams into core

### PR DESCRIPTION
**Pull request by hendrikmuhs**
migrates hlrc dataframe get and _stats to use PageParams, moves PageParams into core for common usage, fix possible NPE in PageParams

Note: The NPE problem is benign because ml clients send page parameters in the body, therefore still non-issue

### CODE REVIEW

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/DataFrameTransformDocumentationIT.java [hunk 1]
```diff
@@ -585,8 +586,7 @@ public void testGetDataFrameTransform() throws IOException, InterruptedException
             // end::get-data-frame-transform-request
 
             // tag::get-data-frame-transform-request-options
-            request.setFrom(0);     // <1>
-            request.setSize(100);   // <2>
+            request.setPageParams(new PageParams(0,100)); // <1>
```
Comment (przemekwitek) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/DataFrameTransformDocumentationIT.java]: Maybe add space after comma (",")?

File: client/rest-high-level/src/test/java/org/elasticsearch/client/DataFrameRequestConvertersTests.java [hunk 2]
```diff
@@ -147,6 +148,24 @@ public void testGetDataFrameTransformStats() {
 
         assertEquals(HttpGet.METHOD_NAME, request.getMethod());
         assertThat(request.getEndpoint(), equalTo("/_data_frame/transforms/foo/_stats"));
+
+        assertFalse(request.getParameters().containsKey("from"));
+        assertFalse(request.getParameters().containsKey("size"));
+
+        getStatsRequest.setPageParams(new PageParams(0, null));
+        request = DataFrameRequestConverters.getDataFrameTransformStats(getStatsRequest);
+        assertEquals("0", request.getParameters().get("from"));
```
Comment (przemekwitek) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/DataFrameRequestConvertersTests.java]: I find the following syntax more readable:
        assertThat(request.getParameters(), hasEntry("from", "0"));

WDYT?

File: client/rest-high-level/src/test/java/org/elasticsearch/client/DataFrameRequestConvertersTests.java [hunk 3]
```diff
@@ -147,6 +148,24 @@ public void testGetDataFrameTransformStats() {
...
+        assertEquals(null, request.getParameters().get("size"));
+
+        getStatsRequest.setPageParams(new PageParams(null, 50));
+        request = DataFrameRequestConverters.getDataFrameTransformStats(getStatsRequest);
+        assertEquals(null, request.getParameters().get("from"));
+        assertEquals("50", request.getParameters().get("size"));
+
+        getStatsRequest.setPageParams(new PageParams(0, 10));
+        request = DataFrameRequestConverters.getDataFrameTransformStats(getStatsRequest);
+        assertEquals("0", request.getParameters().get("from"));
```
Comment (przemekwitek) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/DataFrameRequestConvertersTests.java]: I find the following syntax more readable:
        assertThat(request.getParameters(), allOf(hasEntry("from", "0"), hasEntry("size", "10")));

WDYT?

File: client/rest-high-level/src/test/java/org/elasticsearch/client/DataFrameRequestConvertersTests.java [hunk 4]
```diff
@@ -159,8 +178,17 @@ public void testGetDataFrameTransform() {
         assertFalse(request.getParameters().containsKey("from"));
         assertFalse(request.getParameters().containsKey("size"));
 
-        getRequest.setFrom(0);
-        getRequest.setSize(10);
+        getRequest.setPageParams(new PageParams(0, null));
+        request = DataFrameRequestConverters.getDataFrameTransform(getRequest);
+        assertEquals("0", request.getParameters().get("from"));
```
Comment (przemekwitek) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/DataFrameRequestConvertersTests.java]: I find the following syntax more readable:
assertThat(request.getParameters(), allOf(hasEntry("from", "0"), not(hasKey("size"))));

WDYT?

File: client/rest-high-level/src/main/java/org/elasticsearch/client/core/PageParams.java [hunk 5]
```diff
@@ -57,11 +57,11 @@ public PageParams(@Nullable Integer from, @Nullable Integer size) {
         this.size = size;
     }
 
-    public int getFrom() {
+    public Integer getFrom() {
```
Comment (przemekwitek) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/core/PageParams.java]: FYI, I did the same thing in feature branch:
https://github.com/elastic/elasticsearch/pull/41530/files

**Review by przemekwitek — APPROVED** [review]
LGTM

A few minor comments, mostly around unit tests.

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/1
run elasticsearch-ci/bwc

**Review by benwtrent — APPROVED** [review]
:shipit:

**Pull request merged**